src/test/java/springweb/courseproject/controller/BookControllerTest.java

```diff
@@ -0,0 +1,147 @@
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext context) {
+        mockMvc = webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Sql(scripts = {"classpath:/database/category/add-category.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @Sql(scripts = {"classpath:/database/delete-books-categories-relation.sql"},
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Sql(scripts = {"classpath:/database/book/delete-books.sql"},
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Test
+    @WithMockUser(username = "Admin", roles = "ADMIN")
+    @DisplayName("Create a new book")
+    void createBook_validRequestDto_Success() throws Exception {
+        //Given
+        String jsonRequest = objectMapper.writeValueAsString(CREATE_BOOK_REQUEST_DTO);
+
+        //WHEN
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+        //THEN
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(CREATE_BOOK_RESPONSE_DTO, actual, "id");
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:/database/book/add-book-1.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @WithMockUser(username = "User", roles = "USER")
+    @DisplayName("Get book by id")
+    void getBook_validId_Success() throws Exception {
+        //GIVEN
+        Long validId = 1L;
+        BookDto expected = CREATE_BOOK_RESPONSE_DTO;
+        expected.setIsbn(SECOND_BOOK_ISBN);
+        //WHEN
+        MvcResult result = mockMvc.perform(
+                        get("/books/{id}", validId)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        //THEN
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class);
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual,
+                "id", "categoryIds", "coverImage", "description");
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:/database/book/add-book-2.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @WithMockUser(username = "Admin", roles = "ADMIN")
+    @DisplayName("Delete book by id")
+    void deleteBook_validId_Success() throws Exception {
+        //GIVEN
+        Long validId = 2L;

```

Comment:
Let's use id = 1 wherever we need
```suggestion
        Long validId = 1L;
```